pkg/controller/complianceremediation/complianceremediation_controller.go

```diff
@@ -121,6 +124,21 @@ func (r *ReconcileComplianceRemediation) Reconcile(request reconcile.Request) (r
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileComplianceRemediation) updateDeprecatedMcRemediation(instance *compv1alpha1.ComplianceRemediation, logger logr.Logger) error {
+	remCopy := instance.DeepCopy()
+	remCopy.Spec.Object = remCopy.Spec.MachineConfigContents.DeepCopy()
+	remCopy.Spec.MachineConfigContents = nil
+	return r.client.Update(context.TODO(), remCopy)
```

Comment:
Can you just already continue to reconcileRemediation with the updated object?

Reply:
I couldn't. Cause it would error out anyway cause the object has been updated. So this avoids that.